server/routes/replyRoutes.ts

```diff
@@ -33,18 +36,31 @@ router.get('/:id', async (req, res, next) => {
 
 })
 
-router.post('/', async (req, res, next) => {
+router.post('/', validateReply, async (req, res, next) => {
   try {
-    const reply: Reply = new Reply(req.body.user_id, req.body.post_id, req.body.message);
-    await getRepository(Reply).save(reply);
-    res.status(201).send(reply);
+    console.log(req.params)
+    const user = await getRepository(User).findOne(req.body.userId);
+    if(user) {
+      const post = await getRepository(Post).findOne(req.body.postId);
```

Comment:
Do `getRepository(User).findOne` and `getRepository(Post).findOne` in parallel by using `Promise.all`. More info here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Reply:
I refactored it to use promise.all now and got rid of async. I don't know if the way I implemented it was what you were after, the final product doesn't seem much more cleaner than the start product to be honest! Or maybe it's just my lack of experience in handling promises! If for nothing else, now I at least know a thing or two more about them! :D